airbyte-integrations/connectors/source-close-com/source_close_com/source.py

```diff
@@ -0,0 +1,706 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+from abc import ABC
+from base64 import b64encode
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+from urllib.parse import parse_qsl, urlparse
+
+import requests
+from airbyte_cdk.logger import AirbyteLogger
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+
+
+class CloseComStream(HttpStream, ABC):
+    url_base: str = "https://api.close.com/api/v1/"
+    primary_key: str = "id"
+    number_of_items_per_page: int = 100
+
+    def __init__(self, **kwargs: Mapping[str, Any]):
+        super().__init__(authenticator=kwargs["authenticator"])
+        self.config: Mapping[str, Any] = kwargs
+        self.start_date: str = kwargs["start_date"]
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
```

Comment:
this could do with a docstring explaining exactly how skip and limit work, just for clarity. 

If I understand it correctly, the `_skip` value is the number of _records_ to skip (as opposed to _pages_ or something). `_limit` is how many items were returned in this response. Therefore, for our next_page_token, we take the existing `_skip` value of x records and + the `_limit` value to get total 'seen so far' records meaning that next request starts at the next record. Is this accurate?

Reply:
Yep, you are right. Added docstring and info to bootstrap